tests/unit/components/Player/ItemModal.spec.js

```diff
@@ -4,249 +4,389 @@ import ItemModal from "@/components/Player/ItemModal";
-        itemList: clonedeep(global.dummyItems),
-        itemDetailList: clonedeep(global.dummyItemDetails),
-      },
-    });
-    // test subjective
-    expect(wrapper.vm.currentItem).toEqual(global.dummyItems[0]);
-    expect(wrapper.vm.currentItemDetails).toEqual(global.dummyItemDetails[0]);
-    expect(wrapper.vm.currentItemImage).toBe(global.dummyItemDetails[0].image);
-    expect(wrapper.vm.hasCharLimit).toBe(
-      global.dummyItemDetails[0].has_char_limit
-    );
-    expect(wrapper.vm.maxCharLimit).toBe(
-      global.dummyItemDetails[0].max_char_limit
-    );
-    expect(wrapper.vm.itemType).toBe(global.dummyItems[0].type);
-    expect(wrapper.vm.questionOptions).toStrictEqual(
-      global.dummyItemDetails[0].options
-    );
-    expect(wrapper.vm.questionCorrectAnswer).toBe(0);
-    expect(wrapper.vm.questionText).toBe(global.dummyItemDetails[0].text);
-    expect(wrapper.vm.questionType).toBe(global.dummyItemDetails[0].type);
-    expect(wrapper.vm.isQuestionTypeMCQ).toBe(true);
-    expect(wrapper.vm.isQuestionTypeSubjective).toBe(false);
-
-    // test mcq
-    await wrapper.setProps({
-      selectedItemIndex: 1,
-    });
-
-    expect(wrapper.vm.currentItem).toEqual(global.dummyItems[1]);
-    expect(wrapper.vm.currentItemDetails).toEqual(global.dummyItemDetails[1]);
-    expect(wrapper.vm.currentItemImage).toBe(global.dummyItemDetails[1].image);
-    expect(wrapper.vm.hasCharLimit).toBe(
-      global.dummyItemDetails[1].has_char_limit
-    );
-    expect(wrapper.vm.maxCharLimit).toBe(
-      global.dummyItemDetails[1].max_char_limit
-    );
-    expect(wrapper.vm.itemType).toBe(global.dummyItems[1].type);
-    expect(wrapper.vm.questionOptions).toEqual(
-      global.dummyItemDetails[1].options
-    );
-    expect(wrapper.vm.questionCorrectAnswer).toBe(
-      parseInt(global.dummyItemDetails[1].correct_answer)
-    );
-    expect(wrapper.vm.questionText).toBe(global.dummyItemDetails[1].text);
-    expect(wrapper.vm.questionType).toBe(global.dummyItemDetails[1].type);
-    expect(wrapper.vm.isQuestionTypeMCQ).toBe(true);
-    expect(wrapper.vm.isQuestionTypeSubjective).toBe(false);
-  });
+    const remount = () => {
+      wrapper = mount(ItemModal, {
+        props: {
+          itemList: clonedeep(global.dummyItems),
+          itemDetailList: clonedeep(global.dummyItemDetails),
+        },
+      });
+    };
 
-  it("toggles minimize button", () => {
-    const toggleMinimizeMock = jest.spyOn(ItemModal.methods, "toggleMinimize");
-    const wrapper = mount(ItemModal, {
-      props: {
-        itemList: clonedeep(global.dummyItems),
-        itemDetailList: clonedeep(global.dummyItemDetails),
+    it("should prepare draft responses for each item", () => {
+      expect(wrapper.vm.draftResponses.length).toBe(global.dummyItems.length);
+    });
+
+    it("toggles minimize button", async () => {
+      const toggleMinimizeMock = jest.spyOn(
+        ItemModal.methods,
+        "toggleMinimize"
+      );
+      remount();
```

Comment:
why a remount was needed here?

Reply:
use with BeforeEach and name it as mountWrapper